packages/cozy-stack-client/src/mangoIndex.spec.js

```diff
@@ -162,16 +162,78 @@ describe('getIndexFields', () => {
 })
 
 describe('getIndexNameFromFields', () => {
+  const fields = ['dir_id', 'type', 'name']
+
   it('should return index fields', () => {
-    const fields = ['_id', 'name']
-    expect(getIndexNameFromFields(fields)).toEqual('by__id_and_name')
+    expect(getIndexNameFromFields(fields)).toEqual(
+      'by_dir_id_and_type_and_name'
+    )
   })
 
   it('should return index fields with partial filter', () => {
-    const fields = ['_id', 'name']
-    const partialFilterFields = ['date', 'trashed']
-    expect(getIndexNameFromFields(fields, { partialFilterFields })).toEqual(
-      'by__id_and_name_filter_date_and_trashed'
+    const partialFilter = {
+      trashed: false
+    }
+
+    expect(getIndexNameFromFields(fields, { partialFilter })).toEqual(
+      'by_dir_id_and_type_and_name_filter_(trashed_false)'
+    )
+  })
+
+  it('should return index fields with partial filter with multiple conditions', () => {
+    const partialFilter = {
+      type: 'file',
+      classe: 'image',
+      trashed: true
+    }
+
+    expect(getIndexNameFromFields(fields, { partialFilter })).toEqual(
+      'by_dir_id_and_type_and_name_filter_(type_file)_and_(classe_image)_and_(trashed_true)'
+    )
+  })
+
+  it('should return index fields with partial filter with nested conditions', () => {
+    const partialFilter = {
+      type: 'file',
+      trashed: true,
+      'metadata.notifiedAt': {
+        $exists: false
+      }
+    }
+
+    expect(getIndexNameFromFields(fields, { partialFilter })).toEqual(
+      'by_dir_id_and_type_and_name_filter_(type_file)_and_(trashed_true)_and_(metadata.notifiedAt_$exists_false)'
+    )
+  })
+
+  it('should return index fields with partial filter with $or and $and at root', () => {
+    const partialFilter = {
+      type: 'file',
+      $or: [
+        {
+          trashed: {
+            $exists: false
+          }
+        },
+        {
+          trashed: false
+        }
+      ]
+    }
+
+    expect(getIndexNameFromFields(fields, { partialFilter })).toEqual(
+      'by_dir_id_and_type_and_name_filter_(type_file)_and_((trashed_$exists_false)_$or_(trashed_false))'
+    )
+  })
```

Comment:
Could you add a test with an explicit $or including 3 predicates? :pray:

Reply:
Yes I done it will make `by_dir_id_and_type_and_name_filter_(type_file)_and_((trashed_$exists_false)_$or_(trashed_false)_$or_(class_image))`